distributed/deploy/spec.py

```diff
@@ -371,6 +371,37 @@ def __repr__(self):
             len(self.workers),
         )
 
+    async def _logs(self, workers=None, nanny=False):
+        scheduler, workers = await asyncio.gather(
+            self.scheduler_comm.logs(),
+            self.scheduler_comm.worker_logs(workers=workers, nanny=nanny),
+        )
+        logs = Logs()
+        logs["Scheduler"] = Log("\n".join(line for level, line in scheduler))
+        for k, v in workers.items():
+            logs[k] = Log("\n".join(line for level, line in v))
+
+        return logs
+
+    def logs(self, workers=None, nanny=False):
+        """ Return logs for the scheduler and workers
+
+        Parameters
+        ----------
+        workers: Iterable[str], optional
+            A list of worker addresses to select.
+            Defaults to all workers
+        nanny: bool, optional
+            Get the logs from the nannies rather than the workers
```

Comment:
For implementations that grab logs from the backing system (e.g. kubernetes) this option may not make any sense - the only logs will be stdout/stderr of the process tree (IIRC the nanny and the worker share stdout/stderr). I'd be tempted to remove this option entirely.

Reply:
Done